src/engine/common/JsonSerialization.hxx

```diff
@@ -212,23 +212,13 @@ inline void to_json(json &j, const Point &point)
 inline void to_json(json &j, const MapNodeData &mapNodeData)
 {
   j = json{
-      {"tileID", mapNodeData.tileID},
-      {"tileIndex", mapNodeData.tileIndex},
-      {"origCornerPoint", mapNodeData.origCornerPoint}
-  };
+      {"tileID", mapNodeData.tileID}, {"tileIndex", mapNodeData.tileIndex}, {"origCornerPoint", mapNodeData.origCornerPoint}};
 }
 
 // JSON serializer for MapNode class
-inline void to_json(json &j, const std::unique_ptr<MapNode> &m)
+inline void to_json(json &j, const MapNode &m)
```

Comment:
we should check if there is un-freed memory in this case.

Reply:
I test this, see no leaking. Also not sure why would you expect that, it is constant reference, should cause no issue.